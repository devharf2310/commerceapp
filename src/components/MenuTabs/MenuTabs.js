import React, { PureComponent } from 'react';
import { withRouter } from 'react-router';
import { Tabs, Tab } from 'react-md';
import '../MenuTabs/MenuTabs.scss';

class MenuTabs extends PureComponent {

  handleChange = (activeTabIndex , id) => {
    const { history } = this.props;

    history.replace(`${id}`);
  };

  getActiveTabIndex = () => {
      switch(true){
          case this.isRoutedPath("home"):
            return 0;
          case this.isRoutedPath("products"):
            return 1;
          case this.isRoutedPath("clients"):
            return 2;
          case this.isRoutedPath("contacts"):
            return 3;
        default:
            return 0;
      }
  }

  isRoutedPath = (path) =>  this.props.location.pathname
                            ? this.props.location.pathname.toUpperCase().includes(path.toUpperCase())
                            : false;

    render() {

      return (
        <Tabs
          mobile={false}
          tabId="documentation"
          className="documentation-tabs"
          onTabChange={this.handleChange}
          activeTabIndex={this.getActiveTabIndex()}
        >
          <Tab label="Home"     id="/" key="/" />
          <Tab label="Products" id="/products" key="/products" />
          <Tab label="Clients"  id="/clients" key="/clients" />
          <Tab label="Contact"  id="/contacts" key="/contacts" />
        </Tabs>
      );
    }
  }
  export default withRouter(MenuTabs);
