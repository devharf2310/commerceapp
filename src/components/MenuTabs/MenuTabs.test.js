import React from 'react'
import renderer from 'react-test-renderer'
import MockRouter from 'react-mock-router';
import MenuTabs from './MenuTabs'

describe('Tabs snapshot tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MockRouter><MenuTabs/></MockRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
