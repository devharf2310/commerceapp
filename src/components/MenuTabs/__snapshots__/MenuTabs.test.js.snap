// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs snapshot tests renders correctly 1`] = `
<ul
  className="md-tabs documentation-tabs"
  role="tablist"
  style={
    Object {
      "paddingLeft": undefined,
    }
  }
>
  <span
    aria-hidden={true}
  />
  <li
    aria-controls="documentation-panel-0"
    aria-pressed={false}
    aria-selected={true}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--active"
    id="/"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="tab"
    tabIndex={0}
  >
    <div
      className="md-ink-container"
    />
    <div
      className="md-tab-label"
    >
      Home
    </div>
  </li>
  <li
    aria-controls="documentation-panel-1"
    aria-pressed={false}
    aria-selected={false}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="/products"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="tab"
    tabIndex={0}
  >
    <div
      className="md-ink-container"
    />
    <div
      className="md-tab-label"
    >
      Products
    </div>
  </li>
  <li
    aria-controls="documentation-panel-2"
    aria-pressed={false}
    aria-selected={false}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="/clients"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="tab"
    tabIndex={0}
  >
    <div
      className="md-ink-container"
    />
    <div
      className="md-tab-label"
    >
      Clients
    </div>
  </li>
  <li
    aria-controls="documentation-panel-3"
    aria-pressed={false}
    aria-selected={false}
    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-tab md-tab--inactive"
    id="/contacts"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    role="tab"
    tabIndex={0}
  >
    <div
      className="md-ink-container"
    />
    <div
      className="md-tab-label"
    >
      Contact
    </div>
  </li>
  <span
    className="md-tab-indicator"
    style={
      Object {
        "MozTransform": "translate3d(0px, 0, 0)",
        "WebkitTransform": "translate3d(0px, 0, 0)",
        "height": 2,
        "msTransform": "translate3d(0px, 0, 0)",
        "transform": "translate3d(0px, 0, 0)",
        "width": 160,
      }
    }
  />
</ul>
`;
